diff --git a/node_modules/astro/dist/core/build/generate.js b/node_modules/astro/dist/core/build/generate.js
--- a/node_modules/astro/dist/core/build/generate.js
+++ b/node_modules/astro/dist/core/build/generate.js
@@ -35,7 +35,7 @@
 function createEntryURL(filePath, outFolder) {
-  return new URL("./" + filePath + `?time=${Date.now()}`, outFolder);
+  return new URL("./" + filePath, outFolder);
 }
 async function generatePages(options, internals) {
   const generatePagesTimer = performance.now();

diff --git a/node_modules/astro/dist/core/build/pipeline.js b/node_modules/astro/dist/core/build/pipeline.js
--- a/node_modules/astro/dist/core/build/pipeline.js
+++ b/node_modules/astro/dist/core/build/pipeline.js
@@ -278,7 +278,7 @@
 }
 function createEntryURL(filePath, outFolder) {
-  return new URL("./" + filePath + `?time=${Date.now()}`, outFolder);
+  return new URL("./" + filePath, outFolder);
 }
 function getEntryFilePath(internals, pageData) {
   const id = "\0" + getVirtualModulePageName(ASTRO_PAGE_MODULE_ID, pageData.component);
